index.js

```diff
@@ -143,6 +145,24 @@ function RingPop(options) {
 
     this.destroyed = false;
     this.joiner = null;
+
+    this.startTime = Date.now(); //used for calculating uptime
+
+    this.tchannelVersion = getTchannelVersion(this);
+    this.ringpopVersion = packageJSON.version;
+}
+
+function getTchannelVersion(self)
+{
+    var tchannelPackage = require(path.join(path.dirname(require.resolve('tchannel')), 'package.json'));
+
+    if(tchannelPackage.version !== null) {
+        return tchannelPackage.version;
+    }
+    else {
+        self.logger.error('Could not find: require(path.join(path.dirname(require.resolve(tchannel)), package.json)).version');
```

Comment:
Clean this error message up. Something like `Failed to resolve tchannel version`, {
   packageJsonPath: xxx
})